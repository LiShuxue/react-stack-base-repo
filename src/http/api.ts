import axios from './config';

export interface IApi {
  type: 'get' | 'post' | 'delete' | 'GET' | 'POST' | 'DELETE';
  url: string;
}

// define every api
const getApi: IApi = { type: 'get', url: '/module1/get' };
const postApi: IApi = { type: 'post', url: '/module2/post' };

// export those api as default module, because we alwasy need use those in other place
export default {
  getApi,
  postApi
};

// export those method, then we can use it in bisuness logic
export const testGetMethod = () => {
  return axios({
    method: getApi.type,
    url: getApi.url
  });
};

export const testPostMethod = (data: any) => {
  return axios({
    method: postApi.type,
    url: postApi.url,
    data
  });
};
