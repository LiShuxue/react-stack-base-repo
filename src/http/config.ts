import axios from 'axios';

const instance = axios.create({
  baseURL: process.env.REACT_APP_API_URL
});
instance.defaults.timeout = 20 * 1000;
instance.defaults.headers.post['Content-Type'] = 'application/json; charset=utf-8';

// need to config detail interceptor
instance.interceptors.request.use(
  config => {
    return config;
  },
  error => {
    return Promise.reject(error);
  }
);

// need to config detail interceptor
instance.interceptors.response.use(
  response => {
    return response.data;
  },
  error => {
    return Promise.reject(error);
  }
);

export default instance;
