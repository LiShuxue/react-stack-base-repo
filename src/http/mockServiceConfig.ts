const mockServiceList = ['/module1/get', '/module2/post'];
export default mockServiceList;
