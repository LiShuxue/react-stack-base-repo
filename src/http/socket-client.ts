import socketioClient from 'socket.io-client';

const url: string = process.env.REACT_APP_API_URL || '';
const socket = socketioClient(url);

const start = () => {
  socket.on('connect', () => {
    console.log('client connect to socket server');
  });

  socket.on('message', (data: any) => {
    console.log('receive data from server');
    console.log(data);
    // to do, need storage this data to the redux store
  });

  socket.on('disconnect', () => {
    console.log('client disconnect with server');
  });
};

export default {
  start
};
