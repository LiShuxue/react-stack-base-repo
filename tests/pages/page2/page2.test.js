import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Page2 from 'src/pages/page2';
import { testGetMethod, testPostMethod } from 'src/http/api';

jest.mock('src/http/api', () => ({
  testGetMethod: jest.fn(() => Promise.resolve('get method success')),
  testPostMethod: jest.fn(() => Promise.resolve('post method success'))
}));

test('should render page correctly', () => {
  const { asFragment } = render(<Page2 />);
  expect(asFragment()).toMatchSnapshot();
});

test('should have "testGetApi", "testPostApi" button', () => {
  const { getByText } = render(<Page2 />);

  expect.assertions(2);

  const testGetApi = getByText('testGetApi');
  expect(testGetApi).toBeInTheDocument();

  const testPostApi = getByText('testPostApi');
  expect(testPostApi).toBeInTheDocument();
});

test('click "testGetApi" button should call "testGetMethod"', () => {
  const { getByText } = render(<Page2 />);
  const element = getByText('testGetApi');
  fireEvent.click(element);
  expect(testGetMethod.mock.calls.length).toEqual(1);
});

test('click "testPostApi" button should call "testPostMethod"', () => {
  const { getByText } = render(<Page2 />);
  const element = getByText('testPostApi');
  fireEvent.click(element);
  expect(testPostMethod.mock.calls.length).toEqual(1);
});
