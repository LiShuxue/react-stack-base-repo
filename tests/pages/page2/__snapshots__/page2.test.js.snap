// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page correctly 1`] = `
<DocumentFragment>
  <div>
    <h1>
      page 2
    </h1>
    <hr />
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        testGetApi
      </span>
    </button>
     
    <br />
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        testPostApi
      </span>
    </button>
     
    <br />
    <h2>
      Open Chrome developer tool to inspect your request, and you can also see the response in the console
    </h2>
    <br />
    <h2>
      About websocket relative, wait 20s, server will push data and we will show it in console
    </h2>
  </div>
</DocumentFragment>
`;
